src/csync/csync_exclude.cpp

```diff
@@ -361,6 +361,14 @@ bool ExcludedFiles::isExcludedRemote(const QString &filePath, const QString &bas
         relativePath.chop(1);
     }
 
+    if (excludeHidden) {
+        // Check ther parent directories for . hidden files
+        if (relativePath.startsWith(QLatin1Char('.')) || relativePath.contains(QLatin1String("/."))) {
```

Comment:
Without seeing an example value for `relativePath`, I can't tell whether these checks make sense. I would expect you to calculate the base directory name first before using `startsWith`.
Perhaps the variable name should be updated?

Reply:
The relativePath looks like `foo/bar` or `.foo/.bar`